manifest_staging/charts/secrets-store-csi-driver/templates/crds-upgrade-hook.yaml

```diff
@@ -0,0 +1,67 @@
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: {{ template "sscd.fullname" . }}-upgrade-crds
+{{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+rules:
+  - apiGroups: ["apiextensions.k8s.io"]
+    resources: ["customresourcedefinitions"]
+    verbs: ["get", "create", "update", "patch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: {{ template "sscd.fullname" . }}-upgrade-crds
+{{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+subjects:
+  - kind: ServiceAccount
+    name: {{ template "sscd.fullname" . }}-upgrade-crds
+    namespace: {{ .Release.Namespace }}
+roleRef:
+  kind: ClusterRole
+  name: {{ template "sscd.fullname" . }}-upgrade-crds
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "sscd.fullname" . }}-upgrade-crds
+  namespace: {{ .Release.Namespace }}
+{{ include "sscd.labels" . | indent 2 }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: {{ template "sscd.fullname" . }}-upgrade-crds
+  namespace: {{ .Release.Namespace }}
+  annotations:
+    helm.sh/hook: pre-install,pre-upgrade
+    helm.sh/hook-delete-policy: before-hook-creation,hook-succeeded
+{{ include "sscd.labels" . | indent 2 }}
+spec:
+  serviceAccountName: {{ template "sscd.fullname" . }}-upgrade-crds
+  restartPolicy: OnFailure
+  containers:
+    - name: crds-upgrade
+      image: "{{ .Values.linux.crds.image.repository }}:{{ .Values.linux.crds.image.tag }}"
+      command:
+      - sh
+      - -c
+      - >
+       kubectl apply -f /crds;
+      imagePullPolicy: {{ .Values.linux.crds.image.pullPolicy }}
+  {{- if .Values.imagePullSecrets }}
+  imagePullSecrets:
+    {{ toYaml .Values.imagePullSecrets | indent 4 }}
+  {{- end }}
+  nodeSelector:
+    kubernetes.io/os: linux
```

Comment:
will this not work on windows clusters?

Reply:
We are using this hook Pod to install, upgrade CRDs. Even for windows clusters we can scheduled Pod on master nodes and avoid building/maintaining multiple images for windows.